redux-form/lib/reduxForm.d.ts

```diff
@@ -0,0 +1,551 @@
+     */
+    form: string;
+
+    /**
+     * An adapter function that will render a component based on a string component
+     * type and the props given to a Field. Remember that all you really need to hook
+     * up to your custom component's value and onChange. Defaults to [].
+     *
+     * See Asynchronous Blur Validation Example for more details.
+     */
+    adapter?: (component: string, props: Object) => ReactElement<any>;
+
+    /**
+     * field names for which onBlur should trigger a call to the asyncValidate
+     * function. Defaults to [].
+     *
+     * See Asynchronous Blur Validation Example for more details.
+     */
+    asyncBlurFields?: string[];
+
+    /**
+     * a function that takes all the form values, the dispatch function, and
+     * the props given to your component, and returns a Promise that will
+     * resolve if the validation is passed, or will reject with an object of
+     * validation errors in the form { field1: <String>, field2: <String> }.
+     *
+     * See Asynchronous Blur Validation Example for more details.
+     */
+    asyncValidate?(values: FormData, dispatch: Dispatch<S>, props: P): Promise<any>;
+
+    /**
+     * Whether or not to automatically destroy your form's state in the Redux
+     * store when your component is unmounted. Defaults to true.
+     */
+    destroyOnUnmount?: boolean;
+
+    /**
+     * A function that takes the entire Redux state and returns the state slice
+     * which corresponds to where the redux-form reducer was mounted. This
+     * functionality is rarely needed, and defaults to assuming that the reducer
+     * is mounted under the "form" key.
+     */
+    getFormState?(state: S): any;
+
+    /**
+     * When set to true and enableReinitialize is also set, the form will retain the value of dirty fields when
+     * reinitializing. When this option is not set (the default), reinitializing the form replaces all field values.
+     * This option is useful in situations where the form has live updates or continues to be editable after
+     * form submission; it prevents reinitialization from overwriting user changes. Defaults to false.
+     */
+    keepDirtyOnReinitialize?: boolean;
+
+    /**
+     * If true, implements `shouldComponentUpdate` and shallowly compares _only_
+     * the Redux-connected props that are needed to manage the form state. This
+     * prevents unnecessary updates, but assumes the component is a "pure"
+     * component and does not rely on any input or state other than its props and
+     * the selected Redux store's state. Defaults to true.
+     */
+    pure?: boolean;
+
+    /**
+     * The values with which to initialize your form in componentWillMount().
+     * Particularly useful when Editing Multiple Records, but can also be used
+     * with single-record forms. The values should be in the form
+     * { field1: 'value1', field2: 'value2' }.
+     */
+    initialValues?: FormData;
```

Comment:
You should define a `Partial` type using new typescript 2.1 `keyof`. Otherwise the type check will expect to pass all required properties event if you don't want to specify them initially. 

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/13434/files#diff-92643af80dfab655413729df4df6fd6bR14